api/query/query.go

```diff
@@ -151,3 +152,27 @@ func (qh queryHandler) loadSummary(testRun shared.TestRun) ([]byte, error) {
 func getMemcacheKey(testRun shared.TestRun) string {
 	return "RESULTS_SUMMARY-" + strconv.FormatInt(testRun.ID, 10)
 }
+
+func isRequestCacheable(r *http.Request) bool {
+	ids, err := shared.ParseRunIDsParam(r)
+	return err == nil && len(ids) > 0
+}
+
+func getRequestCacheKey(r *http.Request) interface{} {
```

Comment:
As discussed offline, the burden of this code probably doesn't benefit us much; client-code controls the incoming URLs, and manually typed URLs are unlikely to hit cache anyway.

Let's just use the (full) URL as the cache key for the HTTP cache.